### PR TITLE
Selinux

### PR DESCRIPTION
**Pull request by rhatdan**
Initial patches to get SELinux working with OCID.

### CODE REVIEW

**Comment by rhatdan** [comment]
Currently will not work with passed in SELinux Label.  But if that moves to the sandbox then we can handle it for the initial container.  

Also need some mechanism for reserving all containers that current are using an MCS Label.

**Comment by mrunalp** [comment]
We could temporarily carry my PR to kubernetes if you want because irrespective of the outcome of that discussion there are going to be pod level SELinuxOptions.

**Comment by rhatdan** [comment]
Sure.  Lets get this patch merged and then we can add that as a separate patch.

**Comment by mrunalp** [comment]
make lint is failing.

**Comment by mrunalp** [comment]
@runcom PTAL

File: server/container.go [hunk 1]
```diff
@@ -161,8 +163,12 @@ func (s *Server) createSandboxContainer(containerID string, containerName string
 			options = "ro"
 		}
 
-		//TODO(hmeng): how to use this info? Do we need to handle relabel a FS with Selinux?
-		//selinuxRelabel := mount.GetSelinuxRelabel()
+		if mount.GetSelinuxRelabel() {
+			// Need a way in kubernetes to determine if the volume is shared or private
+			if err := label.Relabel(src, sb.mountLabel, true); err != nil && err != syscall.ENOTSUP {
+				return nil, fmt.Errorf("Relabel failed %s: %q", src, err)
```
Comment (mrunalp) [inline server/container.go]: nit: when errors aren't being logged they should begin with lowercase Relabel -> relabel.

File: server/container.go [hunk 2]
```diff
@@ -240,30 +246,7 @@ func (s *Server) createSandboxContainer(containerID string, containerName string
...
-			}
-
-			level := selinuxOptions.GetLevel()
-			if level == "" {
-				return nil, fmt.Errorf("SELinuxOption.Level is empty")
-			}
-
-			specgen.SetProcessSelinuxLabel(fmt.Sprintf("%s:%s:%s:%s", user, role, t, level))
-		}
+		specgen.SetProcessSelinuxLabel(sb.processLabel)
```
Comment (mrunalp) [inline server/container.go]: We also need to set the mountlabel in the config.

**Comment by rhatdan** [comment]
Who picked the name for these functions?

		specgen.SetProcessSelinuxLabel(sb.processLabel)
		specgen.SetLinuxMountLabel(sb.mountLabel)

**Comment by mrunalp** [comment]
@rhatdan I'll get them fixed in runtime-tools :/

**Comment by runcom** [comment]
LGTM, I'm fine merging this as is if you guys don't want to wait for k8s and have already something

**Comment by mrunalp** [comment]
@runcom Yeah, we can merge this in now and then complete the other bits when k8s PR lands. LGTM.

**Pull request merged**